### PR TITLE
[GH-1876] Add missing bounty information

### LINKED ISSUE #1876: [FEATURE] Fix bounties screens
## 🚀 Feature

We need to implement some changes to fix the current implementation.

**Before starting the game:**

1. When the player doesn't select any task, one of the 3 shown on the screen should be randomly assigned. With the same animation. It is necessary to review if additional time needs to be added to this functionality. 

**In game:**

1. We need to display a pop-up when the player completes the task.

**Game Over:**

1. Remove the completed and failed icons.
2. When the player taps the "Claim" button, the value of the reward corresponding to the completed bounty should appear.
3. A brief animation should display the value of the reward corresponding to the completed bounty. The animation can start from the "Play" button and should end in the gold container.
4. To implement the aforementioned animation, the container, icon, and text corresponding to the GOLD currency need to be added.

### References
Disclaimer: Ignore the character rank progress bar in the references.

Main 
main

Win/Defeat
defeat-cuando claim

defeat

### Assets 

Container: https://drive.google.com/file/d/1v8e2SZUy0dyAhQyAnTSQz67lBPPwX806/view?usp=drive_link
Currencies: https://drive.google.com/drive/folders/1dLVxPHE7KIl-0qI5flx45KsSKSaZr_03?usp=drive_link

### Additional context
Add the container, icon, and text to display the gem currency on the main screen.

### LINKED ISSUE #1942: [BUG] Currency counter starts at 0 when entering Main Screen
https://github.com/user-attachments/assets/f1972bae-16fb-4fd7-ad55-a9268d3468b3

In the video it can be seen that the currency counters start with a value of 0 and then, after a small amount of time it changes to the actual user's value, but this shouldn't happen, the correct value should appear with the screen, this could prove challenging since there is a call to the sever being made for retrieving this value.

### PR DESCRIPTION
**Pull request by tvillegas98**
>[!Warning]
> This pr should be merged alongside [this backend pr](https://github.com/lambdaclass/mirra_backend/pull/733)

>[!Warning]
> The original PR was merged and then reverted (https://github.com/lambdaclass/champions_of_mirra/pull/1877), this is the new PR for merging this branch.

Closes #1876
Closes #1942 

## Motivation
The bounty feature was missing some functionalities and they were implemented in the backend so this pr updates the proto messages.

We needed to the following to the client: 
- Pop up in-game when bounty is completed. 
- Add functionality to the 'claim' button on the deathsplash when bounty is completed. 
- Hide bounties container on deathsplash when player does not complete it. 
- Add animation of the reward when returning to main screen after completing the selected bounty.

Result: 
New main screen design:
Screenshot 2024-07-10 at 16 27 46

Pop up in-game
<video src="https://github.com/lambdaclass/champions_of_mirra/assets/116388676/f83f6ccf-8f97-426f-ac80-392ea00aba04">

Deathsplash when bounty completed, claim button animation and reward animation when returning to main screen after completing the bounty:
<video src="https://github.com/lambdaclass/champions_of_mirra/assets/116388676/a9270544-c1ef-479a-a077-d61b2f719d07">

Deathsplash when bounty not completed
Screenshot 2024-07-05 at 18 05 10

## How has this been tested?
Start a battle and select a bounty, check the following works correctly:
- When completing it the pop-up is shown on the screen
- When completing it you should see the bounty information on the deathsplash, check that the claim button works and gets disabled after claiming
- When returning to main screen after completing a bounty an animation with the bounty reward should come from the play button to the gold container. Check also that the gold amount is being updated correctly
- If the bounty is not completed you should see nothing related to it on the deathsplash

## Checklist
- [x] I have tested the changes locally.
- [x] I have tested the whole game after applying the changes, not only the affected areas.
- [x] I self-reviewed the changes on GitHub, line by line.
- [x] I have tested the changes in another devices.
  - [x] Tested in iOS.
  - [ ] Tested in Android.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by tkz00 — APPROVED** [review]
Works great, tested on my android.

File: client/Assets/Scripts/UI/EndGame/EndGameManager.cs [hunk 1]
```diff
@@ -147,6 +150,29 @@ void ShowMatchInfo()
...
+
+        if (gamePlayer.Player.BountyCompleted) {
+            bounty.SetBounty(GameServerConnectionManager.Instance.bountySelected);
+            bountyContainer.SetActive(true);
+            MainScreenManager.bountyReward = (int)GameServerConnectionManager.Instance.bountySelected.Reward.Amount;
+        }
+    }
+
+    public void ClaimBounty() {
+        ColorUtility.TryParseHtmlString("#666666", out disabledButtonColor);
```
Comment (agustinesco) [inline client/Assets/Scripts/UI/EndGame/EndGameManager.cs:165]: let's be careful with this magic numbers

File: client/Assets/Scripts/UI/EndGame/EndGameManager.cs [hunk 2]
```diff
@@ -18,7 +19,10 @@ public class EndGameManager : MonoBehaviour
     Image rankIcon;
 
     [SerializeField]
-    GameObject defeatedByContainer;
+    GameObject defeatedByContainer,
```
Comment (agustinesco) [inline client/Assets/Scripts/UI/EndGame/EndGameManager.cs:22]: I'm not a big fan of this way of variable declaration, the only thing we achieve is save up some lines over readability

**Review by agustinesco — CHANGES_REQUESTED** [review]
The pr seems to be working as expected! Let's just fix the issue where the bounty overlaps the options buttons and we should be good to go!
image

Also i've left some minor comment's in the code but nothing mandatory

**Review by tkz00 — CHANGES_REQUESTED** [review]
https://github.com/user-attachments/assets/161346d2-2e18-4ee4-b8da-596ee16d4af7

When entering the main screen from anywhere besides the battle scene you have 0 coins, either by changing server or closing and reopening the app, even when going to the character list and selecting a new character.

File: client/Assets/Scripts/UI/EndGame/EndGameManager.cs (hunk 1 above)
Comment (AngieDutra) [reply client/Assets/Scripts/UI/EndGame/EndGameManager.cs:167]: I think that is a color heex value for a button. I agree it could be a variable tho

**Review by tkz00 — CHANGES_REQUESTED** [review]
https://github.com/user-attachments/assets/44bf00d7-38da-4046-9958-f8c4f4a1ba6f

When I go to mainscreen from anywhere I get this small delay of a 0 value in the currency counter, this video was recorded using an ngrok, but maybe using the real servers the problem could be worse.

**Review by agustinesco — APPROVED** [review]
LGTM! playtest it in an android build and it worked as a charm!

**Review by tkz00 — APPROVED** [review]
Great! Worked fine on my android!

**Pull request merged**